### PR TITLE
Append Access Token Hash to ID Token

### PR DESCRIPTION
**Pull request by niwenhao**
## Related issue
#1085 in ory/hydra

## Proposed changes

When request access_token with "grant_type=authorization_code", the id_token of response has not "at_hash" claim. It is optional but FAPI(https://openid.net/specs/openid-financial-api-part-2.html#authorization-response-parameter-injection-attack) use it to prevent some attack.
This PR try to add at_hash.

## Checklist

- [x] I have read the [contributing guidelines](CONTRIBUTING.md)
- [x] I confirm that this pull request does not address a security vulnerability. If this pull request addresses a security
vulnerability, I confirm that I got green light (please contact [hi@ory.sh](mailto:hi@ory.sh)) from the maintainers to push the changes.
- [x] I signed the [Developer's Certificate of Origin](https://github.com/ory/keto/blob/master/CONTRIBUTING.md#developers-certificate-of-origin)
by signing my commit(s). You can amend your signature to the most recent commit by using `git commit --amend -s`. If you
amend the commit, you might need to force push using `git push --force HEAD:<branch>`. Please be very careful when using
force push.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation within the code base (if appropriate)

### CODE REVIEW

File: handler/openid/flow_refresh_token_test.go [hunk 1]
```diff
@@ -157,6 +157,37 @@ func TestOpenIDConnectRefreshHandler_PopulateTokenEndpointResponse(t *testing.T)
 				assert.NotEmpty(t, aresp.GetExtra("id_token"))
```
Comment (aeneasr) [inline handler/openid/flow_refresh_token_test.go]: Let's check here if the hash has been set correctly

File: handler/openid/flow_explicit_token_test.go [hunk 2]
```diff
@@ -113,6 +113,20 @@ func TestExplicit_PopulateTokenEndpointResponse(t *testing.T) {
 				store.EXPECT().GetOpenIDConnectSession(nil, gomock.Any(), areq).AnyTimes().Return(r, nil)
```
Comment (aeneasr) [inline handler/openid/flow_explicit_token_test.go]: We should check here if the hash has been set correctly

File: handler/openid/flow_refresh_token_test.go (hunk 1 above)
Comment (niwenhao) [reply handler/openid/flow_refresh_token_test.go]: Thanks, I will apply it.

File: handler/openid/flow_explicit_token_test.go (hunk 2 above)
Comment (niwenhao) [reply handler/openid/flow_explicit_token_test.go]: Thanks, I will apply it.

**Comment by aeneasr** [comment]
Yeah, you rock, thank you!

**Pull request merged**